payjoin/src/send/mod.rs

```diff
@@ -500,19 +617,19 @@ pub(crate) mod test {
         // Steal 0.5 BTC from the sender output and add it to the receiver output
         proposal.unsigned_tx.output[0].value -= bitcoin::Amount::from_btc(0.5).unwrap();
         proposal.unsigned_tx.output[1].value += bitcoin::Amount::from_btc(0.5).unwrap();
-        ctx.process_proposal(proposal).unwrap();
+        assert!(ctx.process_proposal(proposal.clone()).is_err());
+        Ok(())
     }
 
     #[test]
-    fn test_disable_output_substitution_query_param() {
-        let url =
-            serialize_url(Url::parse("http://localhost").unwrap(), true, None, FeeRate::ZERO, "2")
-                .unwrap();
-        assert_eq!(url, Url::parse("http://localhost?v=2&disableoutputsubstitution=true").unwrap());
-
-        let url =
-            serialize_url(Url::parse("http://localhost").unwrap(), false, None, FeeRate::ZERO, "2")
-                .unwrap();
-        assert_eq!(url, Url::parse("http://localhost?v=2").unwrap());
+    fn test_disable_output_substitution_query_param() -> Result<(), BoxError> {
+        let url = serialize_url(Url::parse("http://localhost")?, true, None, FeeRate::ZERO, "2")
+            .map_err(|e| format!("Failed to serialize url: {e}"))?;
+        assert_eq!(url, Url::parse("http://localhost?v=2&disableoutputsubstitution=true")?);
+
+        let url = serialize_url(Url::parse("http://localhost")?, false, None, FeeRate::ZERO, "2")
+            .map_err(|e| format!("Failed to serialize url: {e}"))?;
```

Comment:
Here and above could be just:

```suggestion
        let url = serialize_url(Url::parse("http://localhost")?, false, None, FeeRate::ZERO, "2")?;
```